docs/docs/general/03-Element Selection/recommended-practices.md

```diff
@@ -4,168 +4,242 @@ title: Recommended Practices
+## Detection Is Flaky or Mislabeled
 
-```javascript
-// Do this
-await aui.click().button('See here').exec();
-await aui.clickText('Sign in');
+### Flaky Detection Across Runs
+Visual detection of elements is highly sensitive to changes in your UI, even if they are small. For example a button that you target may fail randomly on a workflow run every now and then:
 
-// And NOT this
-await aui.click('See here').exec();
-await aui.click('Sign in').exec();
+```typescript
+await aui.click().button().withText('Sign in').exec();
 ```
 
-### Avoid Custom Element Detection if Possible
+You can do three things about this:
 
-If you use `customElement()` you are doing an image-in-image search. Use this sparingly for elements that you can not detect as the execution time is sometimes slower!
+1. Make the element-description more specific, for example with [relational selectors](../../api/04-Relations/above.md)
+2. Add a custom element with `or()` as a fallback
+3. Combine the two methods
 
-```javascript
-// Do this
-await aui.clickButton('Login');
-await aui.clickText('Overview');
+All three of them you can find in the following code:
 
-// And NOT this
-await aui.click().customElement({
-  customImage: '.../login_button.png', 
-  name: 'login button',
-}).exec();
-await aui.click().customElement({
-  customImage: '.../text_overview.png', 
-  name: 'overview button',
-}).exec();
+```typescript
+// Make element-description more specific
+await aui.click().button().withText('Sign in')
+                 .leftOf().button().withText('Login')
+                 .exec();
+
+// Fallback to custom element
+await aui.click().button().withText('Sign in')
+                 .or()
+                 .customElement({
+                     customImage: 'custom_elements/signin.button.png'
+                   })
+                 .exec();
+
+// Add additional element-description and a custom element as fallback
+await aui.click().button().withText('Sign in');
+                 .or().button().containsText('Sign')
+                 .or().customElement({
+                     customImage: 'custom_elements/signin.button.png'
+                   })
+                 .exec();
 ```
 
-## Scrolling
-When you use AskUI you can only interact with elements that you can see on your screen. Therefore you have to scroll down/sideways to interact with currently invisible elements.
+### Icon Detection Fails or Labels Wrong Icon Name
+Detecting icons correctly is difficult because there are millions of icons available. You typically run into two problems:
 
-### Scrolling on Touch Displays
-On touch displays you have to recreate the swipe gesture:
+* Icon is not detected at all
```

Comment:
You mean, detected in one run vs. not detected in another run? As we are still under flaikiness?